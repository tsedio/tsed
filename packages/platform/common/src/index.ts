/**
 * @file Automatically generated by @tsed/barrels.
 */
export * from "./builder/PlatformBuilder.js";
export * from "./config/interfaces/PlatformJsonMapperSettings.js";
export * from "./config/interfaces/PlatformLoggerSettings.js";
export * from "./config/interfaces/PlatformMulterSettings.js";
export * from "./config/interfaces/PlatformRouterSettings.js";
export * from "./config/interfaces/PlatformStaticsSettings.js";
export * from "./config/services/PlatformConfiguration.js";
export * from "./constants/routerOptions.js";
export * from "./decorators/multer/multerFileSize.js";
export * from "./decorators/multer/multerOptions.js";
export * from "./decorators/multer/multipartFile.js";
export * from "./decorators/params/error.js";
export * from "./decorators/params/next.js";
export * from "./decorators/params/request.js";
export * from "./decorators/params/response.js";
export * from "./decorators/PlatformProvider.js";
export * from "./domain/AnyToPromiseWithCtx.js";
export * from "./domain/EndpointMetadata.js";
export * from "./domain/PlatformContext.js";
export * from "./domain/PlatformRouteDetails.js";
export * from "./exports.js";
export * from "./interfaces/AfterInit.js";
export * from "./interfaces/AfterListen.js";
export * from "./interfaces/AfterRoutesInit.js";
export * from "./interfaces/BeforeInit.js";
export * from "./interfaces/BeforeListen.js";
export * from "./interfaces/BeforeRoutesInit.js";
export * from "./interfaces/HandlerType.js";
export * from "./interfaces/IncomingEvent.js";
export * from "./interfaces/OnReady.js";
export * from "./interfaces/OnRequest.js";
export * from "./interfaces/OnResponse.js";
export * from "./interfaces/OnRoutesInit.js";
export * from "./interfaces/PlatformRouteOptions.js";
export * from "./interfaces/ResponseErrorObject.js";
export * from "./interfaces/Route.js";
export * from "./middlewares/PlatformAcceptMimesMiddleware.js";
export * from "./middlewares/PlatformMulterMiddleware.js";
export * from "./services/FakeAdapter.js";
export * from "./services/Platform.js";
export * from "./services/PlatformAdapter.js";
export * from "./services/PlatformApplication.js";
export * from "./services/PlatformHandler.js";
export * from "./services/PlatformMiddlewaresChain.js";
export * from "./services/PlatformRequest.js";
export * from "./services/PlatformResponse.js";
export * from "./services/PlatformTest.js";
export * from "./utils/closeServer.js";
export * from "./utils/createContext.js";
export * from "./utils/createHttpServer.js";
export * from "./utils/createHttpsServer.js";
export * from "./utils/createInjector.js";
export * from "./utils/createServer.js";
export * from "./utils/ensureContext.js";
export * from "./utils/getStaticsOptions.js";
export * from "./utils/listenServer.js";
export * from "./utils/mapReturnedResponse.js";
export * from "./utils/printRoutes.js";
export * from "./utils/registerPlatformAdapter.js";
export * from "./utils/setResponseHeaders.js";
