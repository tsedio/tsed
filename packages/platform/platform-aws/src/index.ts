export * from "./components/PlatformAws";
export * from "./decorators/awsContext";
export * from "./decorators/awsEvent";
