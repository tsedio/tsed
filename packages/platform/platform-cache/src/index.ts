export * from "./interfaces";
export * from "./services/PlatformCache";
export * from "./decorators/useCache";
export * from "./PlatformCacheModule";
