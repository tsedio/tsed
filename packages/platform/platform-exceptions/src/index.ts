// components
export * from "./components/StringErrorFilter";
export * from "./components/ExceptionFilter";
export * from "./components/ErrorFilter";
export * from "./components/MongooseErrorFilter";

// decorators
export * from "./decorators/catch";

// domain
export * from "./domain/ExceptionFiltersContainer";
export * from "./domain/ExceptionSchema";
export * from "./interfaces/ExceptionFilterMethods";

// middlewares
export * from "./services/PlatformExceptions";

// errors
export * from "./errors/ResourceNotFound";
