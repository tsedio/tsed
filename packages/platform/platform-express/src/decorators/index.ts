export * from "./routerSettings";
export * from "./caseSensitive";
export * from "./mergeParams";
export * from "./strict";
