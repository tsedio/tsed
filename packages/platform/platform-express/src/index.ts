/**
 * @file Automatically generated by barrelsby.
 */

export * from "./exports";
export * from "./components/PlatformExpress";
export * from "./decorators/caseSensitive";
export * from "./decorators/mergeParams";
export * from "./decorators/routerSettings";
export * from "./decorators/strict";
export * from "./interfaces/PlatformExpressSettings";
export * from "./interfaces/PlatformExpressStaticsOptions";
export * from "./interfaces/interfaces";
export * from "./middlewares/staticsMiddleware";
