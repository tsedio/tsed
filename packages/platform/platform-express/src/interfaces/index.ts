import Express from "express";
import {PlatformExpressSettings} from "./PlatformExpressSettings";

export * from "./PlatformExpressSettings";
export * from "./PlatformExpressStaticsOptions";

declare global {
  namespace TsED {
    export interface Configuration {
      /**
       * Configuration related to Express platform application.
       */
      express: PlatformExpressSettings;
    }

    export interface NextFunction extends Express.NextFunction {}
  }
}
