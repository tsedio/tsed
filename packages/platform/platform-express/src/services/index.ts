export * from "./PlatformExpressApplication";
export * from "./PlatformExpressHandler";
export * from "./PlatformExpressResponse";
export * from "./PlatformExpressRequest";
export * from "./PlatformExpressRouter";
