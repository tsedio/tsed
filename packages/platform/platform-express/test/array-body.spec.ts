import "@tsed/ajv";
import {BodyParams, Controller, PlatformTest, Post} from "@tsed/common";
import {PlatformTestUtils} from "@tsed/platform-test-utils";
import {CollectionOf, Property, Any} from "@tsed/schema";
import {expect} from "chai";
import SuperTest from "supertest";
import {PlatformExpress} from "../src";
import {rootDir, Server} from "./app/Server";

const utils = PlatformTestUtils.create({
  rootDir,
  platform: PlatformExpress,
  server: Server
});

class MyModel {
  @Property()
  test: string;
}

@Controller("/array")
class TestArrayBodyCtrl {
  @Post("/1")
  async scenario1(@BodyParams() list: any[]) {
    return list;
  }

  @Post("/2")
  async scenario2(@BodyParams() list: number[]) {
    return list;
  }

  @Post("/3")
  async scenario3(@BodyParams() @CollectionOf(Number) list: number[]) {
    return list;
  }

  @Post("/4")
  async scenario4(@BodyParams() @CollectionOf(MyModel) list: MyModel[]) {
    return list;
  }

  @Post("/5")
  async scenario5(@BodyParams() list: any) {
    return list;
  }

  @Post("/6")
  async scenario6(@BodyParams() @Any() list: any) {
    return list;
  }
}

describe("Array Body", () => {
  let request: SuperTest.SuperTest<SuperTest.Test>;

  before(
    utils.bootstrap({
      mount: {
        "/rest": [TestArrayBodyCtrl]
      },
      swagger: [
        {
          path: "/v3/docs",
          specVersion: "3.0.1"
        }
      ]
    })
  );
  after(utils.reset);

  before(() => {
    request = SuperTest(PlatformTest.callback());
  });

  describe("swagger", () => {
    it("should return the swagger", async () => {
      const {body} = await request.get("/v3/docs/swagger.json").expect(200);

      expect(body).to.deep.eq({
        "components": {
          "schemas": {
            "MyModel": {
              "properties": {
                "test": {
                  "type": "string"
                }
              },
              "type": "object"
            }
          }
        },
        "info": {
          "title": "Api documentation",
          "version": "1.0.0"
        },
        "openapi": "3.0.1",
        "paths": {
          "/rest/array/1": {
            "post": {
              "operationId": "testArrayBodyCtrlScenario1",
              "parameters": [],
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "items": {
                        "oneOf": [
                          {
                            "nullable": true,
                            "type": "integer"
                          },
                          {
                            "nullable": true,
                            "type": "number"
                          },
                          {
                            "nullable": true,
                            "type": "string"
                          },
                          {
                            "nullable": true,
                            "type": "boolean"
                          },
                          {
                            "nullable": true,
                            "type": "array"
                          },
                          {
                            "nullable": true,
                            "type": "object"
                          }
                        ]
                      },
                      "type": "array"
                    }
                  }
                },
                "required": false
              },
              "responses": {
                "200": {
                  "description": "Success"
                }
              },
              "tags": [
                "TestArrayBodyCtrl"
              ]
            }
          },
          "/rest/array/2": {
            "post": {
              "operationId": "testArrayBodyCtrlScenario2",
              "parameters": [],
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "items": {
                        "oneOf": [
                          {
                            "nullable": true,
                            "type": "integer"
                          },
                          {
                            "nullable": true,
                            "type": "number"
                          },
                          {
                            "nullable": true,
                            "type": "string"
                          },
                          {
                            "nullable": true,
                            "type": "boolean"
                          },
                          {
                            "nullable": true,
                            "type": "array"
                          },
                          {
                            "nullable": true,
                            "type": "object"
                          }
                        ]
                      },
                      "type": "array"
                    }
                  }
                },
                "required": false
              },
              "responses": {
                "200": {
                  "description": "Success"
                }
              },
              "tags": [
                "TestArrayBodyCtrl"
              ]
            }
          },
          "/rest/array/3": {
            "post": {
              "operationId": "testArrayBodyCtrlScenario3",
              "parameters": [],
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "items": {
                        "type": "number"
                      },
                      "type": "array"
                    }
                  }
                },
                "required": false
              },
              "responses": {
                "200": {
                  "description": "Success"
                }
              },
              "tags": [
                "TestArrayBodyCtrl"
              ]
            }
          },
          "/rest/array/4": {
            "post": {
              "operationId": "testArrayBodyCtrlScenario4",
              "parameters": [],
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "items": {
                        "$ref": "#/components/schemas/MyModel"
                      },
                      "type": "array"
                    }
                  }
                },
                "required": false
              },
              "responses": {
                "200": {
                  "description": "Success"
                }
              },
              "tags": [
                "TestArrayBodyCtrl"
              ]
            }
          },
          "/rest/array/5": {
            "post": {
              "operationId": "testArrayBodyCtrlScenario5",
              "parameters": [],
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "type": "object"
                    }
                  }
                },
                "required": false
              },
              "responses": {
                "200": {
                  "description": "Success"
                }
              },
              "tags": [
                "TestArrayBodyCtrl"
              ]
            }
          },
          "/rest/array/6": {
            "post": {
              "operationId": "testArrayBodyCtrlScenario6",
              "parameters": [],
              "requestBody": {
                "content": {
                  "application/json": {
                    "schema": {
                      "oneOf": [
                        {
                          "nullable": true,
                          "type": "integer"
                        },
                        {
                          "nullable": true,
                          "type": "number"
                        },
                        {
                          "nullable": true,
                          "type": "string"
                        },
                        {
                          "nullable": true,
                          "type": "boolean"
                        },
                        {
                          "nullable": true,
                          "type": "array"
                        },
                        {
                          "nullable": true,
                          "type": "object"
                        }
                      ]
                    }
                  }
                },
                "required": false
              },
              "responses": {
                "200": {
                  "description": "Success"
                }
              },
              "tags": [
                "TestArrayBodyCtrl"
              ]
            }
          }
        },
        "tags": [
          {
            "name": "TestArrayBodyCtrl"
          }
        ]
      });
    });
  });

  describe("POST /rest/array/1", () => {
    it("should return list", async () => {
      const {body} = await request.post("/rest/array/1")
        .send([{test: "test"}])
        .expect(200);

      expect(body).to.deep.eq([
        {
          "test": "test"
        }
      ]);
    });
  });
  describe("POST /rest/array/2", () => {
    it("should return list", async () => {
      const {body} = await request.post("/rest/array/2")
        .send([{test: 1}])
        .expect(200);

      expect(body).to.deep.eq([
        {
          "test": 1
        }
      ]);
    });
  });

  describe("POST /rest/array/3", () => {
    it("should return list", async () => {
      const {body} = await request.post("/rest/array/3")
        .send([1])
        .expect(200);

      expect(body).to.deep.eq([
        1
      ]);
    });
  });

  describe("POST /rest/array/4", () => {
    it("should return list", async () => {
      const {body} = await request.post("/rest/array/4")
        .send([{test: "1"}])
        .expect(200);

      expect(body).to.deep.eq([
        {
          "test": "1"
        }
      ]);
    });
  });
  describe("POST /rest/array/5", () => {
    it("should throw a bad request", async () => {
      const {body} = await request.post("/rest/array/5")
        .send([{test: "1"}])
        .expect(400);

      expect(body).to.deep.eq({
        "errors": [
          {
            "data": [
              {
                "test": "1"
              }
            ],
            "dataPath": "",
            "keyword": "type",
            "instancePath": "",
            "message": "must be object",
            "params": {
              "type": "object"
            },
            "schemaPath": "#/type"
          }
        ],
        "message": "Bad request on parameter \"request.body\".\nValue must be object. Given value: [{\"test\":\"1\"}]",
        "name": "AJV_VALIDATION_ERROR",
        "status": 400
      });
    });
  });
});
