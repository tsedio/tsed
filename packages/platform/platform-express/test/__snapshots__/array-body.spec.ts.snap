// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array Body swagger should return the swagger (OS3) 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "MyModel": Object {
        "properties": Object {
          "test": Object {
            "type": "string",
          },
        },
        "type": "object",
      },
    },
  },
  "info": Object {
    "title": "Api documentation",
    "version": "1.0.0",
  },
  "openapi": "3.0.1",
  "paths": Object {
    "/rest/array/1": Object {
      "post": Object {
        "operationId": "testArrayBodyCtrlScenario1",
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "items": Object {
                  "nullable": true,
                  "oneOf": Array [
                    Object {
                      "type": "integer",
                    },
                    Object {
                      "type": "number",
                    },
                    Object {
                      "type": "string",
                    },
                    Object {
                      "type": "boolean",
                    },
                    Object {
                      "type": "array",
                    },
                    Object {
                      "type": "object",
                    },
                  ],
                },
                "type": "array",
              },
            },
          },
          "required": false,
        },
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "TestArrayBodyCtrl",
        ],
      },
    },
    "/rest/array/2": Object {
      "post": Object {
        "operationId": "testArrayBodyCtrlScenario2",
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "items": Object {
                  "nullable": true,
                  "oneOf": Array [
                    Object {
                      "type": "integer",
                    },
                    Object {
                      "type": "number",
                    },
                    Object {
                      "type": "string",
                    },
                    Object {
                      "type": "boolean",
                    },
                    Object {
                      "type": "array",
                    },
                    Object {
                      "type": "object",
                    },
                  ],
                },
                "type": "array",
              },
            },
          },
          "required": false,
        },
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "TestArrayBodyCtrl",
        ],
      },
    },
    "/rest/array/3": Object {
      "post": Object {
        "operationId": "testArrayBodyCtrlScenario3",
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "items": Object {
                  "type": "number",
                },
                "type": "array",
              },
            },
          },
          "required": false,
        },
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "TestArrayBodyCtrl",
        ],
      },
    },
    "/rest/array/4": Object {
      "post": Object {
        "operationId": "testArrayBodyCtrlScenario4",
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "items": Object {
                  "$ref": "#/components/schemas/MyModel",
                },
                "type": "array",
              },
            },
          },
          "required": false,
        },
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "TestArrayBodyCtrl",
        ],
      },
    },
    "/rest/array/5": Object {
      "post": Object {
        "operationId": "testArrayBodyCtrlScenario5",
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "type": "object",
              },
            },
          },
          "required": false,
        },
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "TestArrayBodyCtrl",
        ],
      },
    },
    "/rest/array/6": Object {
      "post": Object {
        "operationId": "testArrayBodyCtrlScenario6",
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "application/json": Object {
              "schema": Object {
                "nullable": true,
                "oneOf": Array [
                  Object {
                    "type": "integer",
                  },
                  Object {
                    "type": "number",
                  },
                  Object {
                    "type": "string",
                  },
                  Object {
                    "type": "boolean",
                  },
                  Object {
                    "type": "array",
                  },
                  Object {
                    "type": "object",
                  },
                ],
              },
            },
          },
          "required": false,
        },
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "TestArrayBodyCtrl",
        ],
      },
    },
  },
  "tags": Array [
    Object {
      "name": "TestArrayBodyCtrl",
    },
  ],
}
`;
